### DISCUSSION TITLE
Do you have any example with getting request from the server and rendering the content as a list of components?

### DISCUSSION THREAD

**Discussion by the-man-with-a-golden-mind** [vanjs-org/van#203] [Q&A]
Hi there. 
I am getting list of Posts from my local server. To be honest something which is really easy to do in React in VanJs is quite tricky. 
Lets say that I am getting : [ {name: "MM"}, {name: "NN"}] from server. 
I would like to show "Loading" on the screen when the request is processed and then render those "Posts" (one after another_ with some HTML structures around. Can not figure it out. 

Best,
MM

**Comment by Tao-VanJS** [comment]
I don't think this is more difficult to implement with VanJS than React. Could you show me the code with React and maybe I can demonstrate the equivalent code with VanJS?

**Reply by the-man-with-a-golden-mind** [reply]
How would you implement useEffect in vanJs?

**Reply by Tao-VanJS** [reply]
In VanJS, you can use `van.derive` for side effects, see https://vanjs.org/tutorial#side-effect.

**Reply by the-man-with-a-golden-mind** [reply]
Thank YOU!
